app/src/main/java/hm/orz/chaos114/android/tumekyouen/db/KyouenDb.java

```diff
@@ -14,402 +7,406 @@
+                model.setStageCount(count);
+                int clearCount = cursor.getInt(1);
+                model.setClearStageCount(clearCount);
+            }
+
+            return model;
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+            if (database != null) {
+                database.close();
+            }
+        }
+    }
+
+    /**
+     * クリア済みのステージリストを取得する。
+     *
+     * @return クリア済みステージリスト
+     */
+    public List<TumeKyouenModel> selectAllClearStage() {
+        SQLiteDatabase database = null;
+        Cursor cursor = null;
+        try {
+            database = mHelper.getReadableDatabase();
+
+            // レコード取得
+            cursor = database.query(TABLE_NAME, null,
+                    TumeKyouenDataColumns.CLEAR_FLAG + " = 1", null, null,
+                    null, null, null);
+
+            List<TumeKyouenModel> list = new ArrayList<TumeKyouenModel>();
+            while (cursor.moveToNext()) {
+                list.add(cursorToTumeKyouenModel(cursor));
+            }
+
+            return list;
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+            if (database != null) {
+                database.close();
+            }
+        }
+    }
+
+    /**
+     * クリア情報を同期する。登録されていないステージ情報は無視する。
+     *
+     * @param clearList クリア情報リスト
+     */
+    public void updateSyncClearData(List<TumeKyouenModel> clearList) {
+        for (TumeKyouenModel model : clearList) {
+            TumeKyouenModel dbModel = selectCurrentStage(model.getStageNo());
+            if (dbModel == null) {
+                continue;
+            }
+            dbModel.setClearFlag(TumeKyouenModel.CLEAR);
+            dbModel.setClearDate(model.getClearDate());
+            updateClearFlag(dbModel, model.getClearDate());
+        }
+    }
+
+    /** カラム名 */
+    public interface TumeKyouenDataColumns extends BaseColumns {
+        /** ステージ番号 */
+        String STAGE_NO = "stage_no";
+
+        /** サイズ */
+        String SIZE = "size";
+
+        /** ステージ上の石の配置 */
+        String STAGE = "stage";
+
+        /** 作者 */
+        String CREATOR = "creator";
+
+        /** クリアフラグ */
+        String CLEAR_FLAG = "clear_flag";
+
+        /** クリア日付 */
+        String CLEAR_DATE = "clear_date";
+    }
+
+    /**
+     * パズル情報DBのヘルパークラス
+     *
+     * @author noboru
+     */
+    class KyouenDbOpenHelper extends SQLiteOpenHelper {
+
+        /** DBのバージョン */
+        private static final int DB_VERSION = 2;
+
+        /** DB名 */
+        private static final String DB_NAME = "irokae.db";
+
+        private KyouenDbOpenHelper(Context context) {
+            super(context, DB_NAME, null, DB_VERSION);
+        }
+
+        @Override
+        public void onCreate(SQLiteDatabase db) {
+            StringBuilder sql = new StringBuilder();
+            sql.append("CREATE TABLE IF NOT EXISTS ");
+            sql.append(TABLE_NAME);
```

Comment:
WARNING: [ConsecutiveAppendsShouldReuse] StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.
http://pmd.sourceforge.net/pmd-5.2.3/pmd-java/rules/java/strings.html#ConsecutiveAppendsShouldReuse